Midichlorians/Dev/Beta2/MidichloriansDataDictionary.h

```diff
@@ -0,0 +1,142 @@
+
+#ifdef COMPILE_MIDICHLORIANS_DATA_DICTIONARY_FOR_IOS_CLIENT
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+#endif  // COMPILE_MIDICHLORIANS_DATA_DICTIONARY_FOR_IOS_CLIENT
+
+namespace current {
+namespace midichlorians {
+
+// Base event.
+CURRENT_STRUCT(MidichloriansBaseEvent) {
+  CURRENT_FIELD(user_ms, std::chrono::milliseconds);
+  CURRENT_FIELD(device_id, std::string);
+  CURRENT_FIELD(client_id, std::string);
+};
+
+// iOS-specific base event.
+CURRENT_STRUCT(iOSBaseEvent, MidichloriansBaseEvent) {
+  // TODO: do we need this field?
+  CURRENT_FIELD(description, std::string);
+};
+
+// iOS event denoting the very first application launch.
+// Emitted once during Midichlorians initialization.
+CURRENT_STRUCT(iOSFirstLaunchEvent, iOSBaseEvent){};
+
+// iOS event with the detailed application information.
+// Emitted once during Midichlorians initialization.
+CURRENT_STRUCT(iOSAppLaunchEvent, iOSBaseEvent) {
+  CURRENT_FIELD(binary_version, std::string);
+  CURRENT_FIELD(cf_version, std::string);
+  CURRENT_FIELD(app_install_time, uint64_t);
+  CURRENT_FIELD(app_update_time, uint64_t);
+  CURRENT_DEFAULT_CONSTRUCTOR(iOSAppLaunchEvent) {}
+  CURRENT_CONSTRUCTOR(iOSAppLaunchEvent)(
+      const std::string& cf_version, uint64_t app_install_time, uint64_t app_update_time)
+      : binary_version(__DATE__ " " __TIME__),
+        cf_version(cf_version),
+        app_install_time(app_install_time),
+        app_update_time(app_update_time) {
+    iOSBaseEvent::description = "*FinishLaunchingWithOptions";
+  }
+};
+
+// iOS event with the detailed device information.
+// Emitted once during Midichlorians initialization.
+CURRENT_STRUCT(iOSDeviceInfo, iOSBaseEvent) { CURRENT_FIELD(info, (std::map<std::string, std::string>)); };
+
+// iOS event emitted by `identify` method.
+CURRENT_STRUCT(iOSIdentifyEvent, iOSBaseEvent){};
+
+// iOS event emitted by `focusEvent` method.
+CURRENT_STRUCT(iOSFocusEvent, iOSBaseEvent) {
+  CURRENT_FIELD(gained_focus, bool);
+  CURRENT_FIELD(source, std::string);
+  CURRENT_CONSTRUCTOR(iOSFocusEvent)(bool gained_focus = false, const std::string& source = "")
+      : gained_focus(gained_focus), source(source) {}
+};
+
+CURRENT_STRUCT(iOSGenericEvent, iOSBaseEvent) {
+  CURRENT_FIELD(event, std::string);
+  CURRENT_FIELD(source, std::string);
+  CURRENT_FIELD(fields, (std::map<std::string, std::string>));
+  // TODO: do we need this separation to `fields` + `complex_fields`?
+  CURRENT_FIELD(complex_fields, (std::map<std::string, std::string>));
+  CURRENT_FIELD(unparsable_fields, std::vector<std::string>);
+  CURRENT_DEFAULT_CONSTRUCTOR(iOSGenericEvent) {}
+#ifdef COMPILE_MIDICHLORIANS_DATA_DICTIONARY_FOR_IOS_CLIENT
+  CURRENT_CONSTRUCTOR(iOSGenericEvent)(NSString * input_event, NSString * input_source, NSDictionary * input) {
+    // A somewhat strict yet safe way to parse dictionaries. Imperfect but works. -- D. K.
+    event = [input_event UTF8String];
```

Comment:
Theoretically yes, but it means some weird and incorrect usage of our API. Maybe we should ignore such kind of call and `NSLog` this fact?